### PR TITLE
src: fix/add some

### PR DESCRIPTION
**Pull request by idinium96**
will add more if found more.

### CODE REVIEW

File: src/classes/MyHandler.ts [hunk 1]
```diff
@@ -457,10 +457,15 @@ export = class MyHandler extends Handler {
...
+                        const sellPrice = match.sell.toValue(keyPrice.metal);
+                        const minimumKeysDupeCheck = this.minimumKeysDupeCheck * keyPrice.toValue();
 
                         if (
-                            item.effect !== null &&
-                            match.buy.toValue(keyPrice.metal) > this.minimumKeysDupeCheck * keyPrice.toValue()
+                            (item.effect !== null || item.australium === true) && // check on only items with effects or australium
+                            buying && // check only items on their side
+                            (buyPrice > minimumKeysDupeCheck || sellPrice > minimumKeysDupeCheck)
+                            // if their side contains invalid_items, will use our side value
```
Comment (Nicklason) [inline src/classes/MyHandler.ts:468]: Why only australiums and items with effects?

File: src/classes/UserCart.ts [hunk 2]
```diff
@@ -36,7 +36,12 @@ class UserCart extends Cart {
 
         // TODO: Check for dupes
 
-        if ((this.bot.handler as MyHandler).hasDupeCheckEnabled()) {
+        const isDupedCheckEnabled = (this.bot.handler as MyHandler).hasDupeCheckEnabled();
+        const keyPrice = this.bot.pricelist.getKeyPrice();
+        const theirItemsValue = this.getTheirCurrencies().toValue(keyPrice.metal);
+        const minimumKeysDupeCheck = (this.bot.handler as MyHandler).getMinimumKeysDupeCheck() * keyPrice.toValue();
+
+        if (isDupedCheckEnabled && theirItemsValue > minimumKeysDupeCheck) {
```
Comment (Nicklason) [inline src/classes/UserCart.ts:44]: And here you don't do the check for australiums / effects?

File: src/classes/Trades.ts [hunk 3]
```diff
@@ -738,7 +738,7 @@ export = class Trades {
 
             offer.data('finishTimestamp', finishTimestamp);
 
-            const processTime = finishTimestamp - offer.data('handleTimeStamp');
+            const processTime = finishTimestamp - offer.data('handleTimestamp');
```
Comment (Nicklason) [inline src/classes/Trades.ts:741]: Nice haha

File: src/classes/MyHandler.ts (hunk 1 above)
Comment (idinium96) [reply src/classes/MyHandler.ts:468]: Previously it was only `item.effect !== null`. What else that you think the best items to check for duped?

File: src/classes/UserCart.ts (hunk 2 above)
Comment (idinium96) [reply src/classes/UserCart.ts:44]: Well yes, I should add that but I am not sure how.

File: src/classes/Trades.ts (hunk 3 above)
Comment (idinium96) [reply src/classes/Trades.ts:741]: Yea, I am surprised no one realized that. It's frustrating to see it always say unknown ms so I take a look and found this.

File: src/classes/MyHandler.ts (hunk 1 above)
Comment (idinium96) [reply src/classes/MyHandler.ts]: I have changed it to check on all items that have > minimum keys to check

File: src/classes/Commands.ts [hunk 4]
```diff
@@ -1843,6 +1843,39 @@ export = class Commands {
             item.quality = quality;
         }
 
+        if (params.craftable !== undefined) {
+            if (typeof params.craftable !== 'boolean') {
+                this.bot.sendMessage(steamID, `Craftable must be "true" or "false" only.`);
+                return null;
+            }
+            if (params.craftable === false) {
```
Comment (lexidor) [inline src/classes/Commands.ts:1851]: Expression can be simplified.
```TS
item.craftable = params.craftable;
```

File: src/classes/Commands.ts [hunk 5]
```diff
@@ -1843,6 +1843,39 @@ export = class Commands {
...
+                item.craftable = true;
+            }
+        }
+
+        if (params.australium !== undefined) {
+            if (typeof params.australium !== 'boolean') {
+                this.bot.sendMessage(steamID, `Australium must be "true" or "false" only.`);
+                return null;
+            }
+            if (params.australium === false) {
```
Comment (lexidor) [inline src/classes/Commands.ts:1863]: Expression can be simplified.

File: src/classes/Commands.ts [hunk 6]
```diff
@@ -1843,6 +1843,39 @@ export = class Commands {
...
+                item.australium = false;
+            } else {
+                item.australium = true;
+            }
+        }
+
+        if (params.killstreak !== undefined) {
+            const killstreak = parseInt(params.killstreak);
+            if (isNaN(killstreak) || killstreak > 3) {
```
Comment (lexidor) [inline src/classes/Commands.ts:1872]: Check is not sufficient.
`parseInt()` may return a negative number.

File: src/classes/MyHandler.ts [hunk 7]
```diff
@@ -238,12 +237,12 @@ export = class MyHandler extends Handler {
...
         log.debug('Group relation changed', { steamID: groupID, relationship: relationship });
         if (relationship === SteamUser.EClanRelationship.Invited) {
-            const join = !this.groups.includes(groupID.getSteamID64());
+            const join = this.groups.includes(groupID.getSteamID64());
 
             log.info(
                 'Got invited to group ' + groupID.getSteamID64() + ', ' + (join ? 'accepting...' : 'declining...')
             );
-            this.bot.client.respondToGroupInvite(groupID, !this.groups.includes(groupID.getSteamID64()));
+            this.bot.client.respondToGroupInvite(groupID, this.groups.includes(groupID.getSteamID64()));
```
Comment (poespas) [inline src/classes/MyHandler.ts:245]: use the `join` constant you have just created instead of `this.groups.includes(groupID.getSteamID64())`

Comment (poespas) [reply src/classes/MyHandler.ts:245]: also maybe don't call the variable `join`, but something along the lines of `shouldJoin` to confuse possible confusion.

**Pull request closed without merging**